extensions/oplogPopulator/modules/ConnectorsManager.js

```diff
@@ -296,14 +294,56 @@ class ConnectorsManager {
                 });
             }
         });
-        const logMessage = connectorUpdateFailed ? 'Failed to update some or all the connectors' :
-            'Successfully updated all the connectors';
-        const logFunction = connectorUpdateFailed ? this._logger.error.bind(this._logger) :
-            this._logger.info.bind(this._logger);
-        logFunction(logMessage, {
-            method: 'ConnectorsManager._updateConnectors',
-            connectorsStatus,
-        });
+        if (Object.keys(connectorsStatus).length > 0) {
+            this._logger.info('Successfully updated connectors', {
+                method: 'ConnectorsManager._updateConnectors',
+                connectorsStatus,
+            });
+        }
+    }
+
+    /**
+     * Checks if connectors are in a failed state and restarts them
+     * @param {connector} connector connector instance
+     * @returns {Promise<undefined>} undefined
+     * @throws {InternalError}
+     */
+    async _validateConnectorState(connector) {
+        if (!connector.isRunning) {
+            return;
+        }
+
+        try {
+            const connectorStatus = await this._kafkaConnect.getConnectorStatus(connector.name);
+            const isConnectorFailed = connectorStatus?.connector?.state === 'FAILED';
+            const areTasksFailed = connectorStatus?.tasks?.some(task => {
+                if (task.state === 'FAILED') {
```

Comment:
what is the state during "startup"?
is there no "intermediate" state where we could restart the task inappropriately?

Reply:
Doesn't look like there is an intermediate state, the only states possible according to [kafka connect docs](https://docs.confluent.io/platform/current/connect/monitoring.html#:~:text=The%20following%20states%20are%20possible,task%20has%20been%20administratively%20paused.) are : 

- UNASSIGNED: The connector/task has not yet been assigned to a worker.
- RUNNING: The connector/task is running.
- PAUSED: The connector/task has been administratively paused.
- FAILED: The connector/task has failed (usually by raising an exception, which is reported in the status output).